src/services/index.ts

```diff
@@ -97,3 +100,47 @@ const buildSingleDataEntry = (messages: Array<Message>): DataEntry => {
 //4.
 
 
+// send transaction with hash constructed from batch of messages
+const unichatAddress = "0x47a1fc3ef4d00e9862724e4d2cbc96a35e26a803"; // TODO: update this to other networks as needed
+
+// The ERC-20 Contract ABI, which is a common contract interface
+// for tokens (this is the Human-Readable ABI format)
+const unichatAbi = [
+  // human-readable ABI for Unichat contract
+  "function AddData(string memory _oldHash, string memory _newHash) payable public returns(bool)",
+  
+  "function WithdrawFunds(address payable _address) external returns(bool)",
+
+  "event NewEntry(string _hash)",
+  
+];
+
+const provider = new ethers.providers.JsonRpcProvider(process.env.URL);
+
+const signer = provider.getSigner()
+
+// The Contract object
+const unichatContract = new ethers.Contract(unichatAddress, unichatAbi, provider);
+
+const invokeAddData = (oldHash: string, newHash: string) => {
+
+    const unichatWithSigner = unichatContract.connect(signer);
+
+    const eth = ethers.utils.parseUnits("1.0", 18);
+
+    const tx = unichatWithSigner.AddData(oldHash, newHash);
+
+    return {
+        oldHash // TODO: determine better return
+    }
+}
+
+const listenForEvents = () => {
+    unichatContract.on("NewEntry", (hash, event) => {
+        console.log(`New hash received: ${ hash}`);
```

Comment:
the function we want to execute when a new hash is received, is the function we should create and define here in the services. Something like `onNewEntry`